lib/P6doc.pm6

```diff
@@ -1,187 +1,356 @@
-			$final-docee = %prefixes{ $<choice>.lc };
-			last;
-		} else {
-			say "$word doesn't seem to apply here.\n";
-			next;
-		}
-	}
+    if %prefixes.values.grep( -> @o { +@o > 1 } ) {
+        my (%indexes,$base-idx);
+        $base-idx = 0;
+        for %options.kv -> $k,@o {
+            %indexes.push: @o>>.map({ ++$base-idx }) Z=> @(%found{$k});
+        }
+        %prefixes = %indexes;
+    }
+
+    my $prompt-text = "Narrow your choice? ({ %prefixes.keys.sort.join(', ') }, or !{ '/' ~ 'q' if !%prefixes<q> } to quit): ";
+
+    while prompt($prompt-text).words -> $word {
+        if $word  ~~ '!' or ($word ~~ 'q' and !%prefixes<q>) {
+            exit 1;
+        } elsif $word ~~ /:i $<choice> = [ @(%prefixes.keys) ] / {
+            $final-docee = %prefixes{ $<choice>.lc };
+            last;
+        } else {
+            say "$word doesn't seem to apply here.\n";
+            next;
+        }
+    }
 
-	return $final-docee;
+    return $final-docee;
 }
 
 sub locate-curli-module($module) {
-	my $cu = try $*REPO.need(CompUnit::DependencySpecification.new(:short-name($module)));
-	unless $cu.DEFINITE {
-		fail "No such type '$module'";
-		#exit 1;
-	}
-	return ~ $cu.repo.prefix.child('sources/' ~ $cu.repo-id);
+    my $cu = try $*REPO.need(CompUnit::DependencySpecification.new(:short-name($module)));
+    unless $cu.DEFINITE {
+        fail "No such type '$module'";
+        #exit 1;
+    }
+    return ~ $cu.repo.prefix.child('sources/' ~ $cu.repo-id);
 }
 
 # see: Zef::Client.list-installed()
 # Eventually replace with CURI.installed()
 # https://github.com/rakudo/rakudo/blob/8d0fa6616bab6436eab870b512056afdf5880e08/src/core/CompUnit/Repository/Installable.pm#L21
 sub list-installed() is export {
-	my @curs       = $*REPO.repo-chain.grep(*.?prefix.?e);
-	my @repo-dirs  = @curs>>.prefix;
-	my @dist-dirs  = |@repo-dirs.map(*.child('dist')).grep(*.e);
-	my @dist-files = |@dist-dirs.map(*.IO.dir.grep(*.IO.f).Slip);
-
-	my $dists := gather for @dist-files -> $file {
-		if try { Distribution.new( |%(from-json($file.IO.slurp)) ) } -> $dist {
-			my $cur = @curs.first: {.prefix eq $file.parent.parent}
-			my $dist-with-prefix = $dist but role :: { has $.repo-prefix = $cur.prefix };
-			take $dist-with-prefix;
-		}
-	}
+    my @curs       = $*REPO.repo-chain.grep(*.?prefix.?e);
+    my @repo-dirs  = @curs>>.prefix;
+    my @dist-dirs  = |@repo-dirs.map(*.child('dist')).grep(*.e);
+    my @dist-files = |@dist-dirs.map(*.IO.dir.grep(*.IO.f).Slip);
+
+    my $dists := gather for @dist-files -> $file {
+        if try { Distribution.new( |%(from-json($file.IO.slurp)) ) } -> $dist {
+            my $cur = @curs.first: {.prefix eq $file.parent.parent}
+            my $dist-with-prefix = $dist but role :: { has $.repo-prefix = $cur.prefix };
+            take $dist-with-prefix;
+        }
+    }
+}
+
+###
+### NEXT
+###
+
+#| Create a Perl6::Documentable::Registry for the given directory
+sub compose-registry(
```

Comment:
This is a style issue, so this may be disregarded in my opinion. I think this signature would be clearer if written like this.

    sub compose-registry (
        $topdir,
        @dirs = ['type'],
        --> Perl6::Documentable::Registry
    ) {
        ...
    }

Here, every line contains one element of the signature, whereas your code has two parts, but only on the last line, which feels inconsistent. Additionally, using this style, it's easier to attach POD6 to the individual elements. (The `#|` POD style you have for the subroutine name can also be used for individual arguments.)